cmd.go

```diff
@@ -37,79 +38,152 @@ import (
-			log.WithError(err).Fatal("could not create variables")
+// Runner represents a command runner
+type Runner struct {
+	prefix      string
+	runner      interface{}
+	cfg         interface{}
+	prevPreRunE func(cmd *cobra.Command, args []string) error
+}
+
+// NewRunner creates a new Runner
+func NewRunner(prefix string, runner interface{}, cfg interface{}) *Runner {
+	return &Runner{
+		prefix: prefix,
+		runner: runner,
+		cfg:    cfg,
+	}
+}
+
+// SetupRoot sets up the root Command
+func (r *Runner) SetupRoot(cmd *cobra.Command) *Runner {
+	if cmd.PersistentFlags().Lookup("env") == nil {
+		cmd.PersistentFlags().String("env", strings.ToLower(r.Getenv("environment")), "specifies the environment")
+	}
+	if cmd.PersistentFlags().Lookup("loglevel") == nil {
+		cmd.PersistentFlags().String("loglevel", strings.ToLower(r.Getenv("loglevel")), "specifies the log level")
+	}
+	if cmd.PersistentFlags().Lookup("config") == nil {
+		cmd.PersistentFlags().StringSlice("config", nil, "specifies a .env configuration file to load")
+	}
+
+	r.prevPreRunE = cmd.PersistentPreRunE
+
+	if r.prevPreRunE == nil {
+		r.prevPreRunE = func(cmd *cobra.Command, args []string) error {
+			return nil
 		}
+	}
+
+	cmd.PersistentPreRunE = r.preRunE
+	return r
+}
 
-		sort.Strings(vars)
-		for _, v := range vars {
-			if strings.Contains(v, "=#") {
-				fmt.Println("#" + v)
-			} else {
-				fmt.Println(v)
+// Setup sets up the Command
+func (r *Runner) Setup(cmd *cobra.Command) *Runner {
+	if cmd.RunE == nil {
+		cmd.RunE = r.runE
+	}
+
+	cmd.AddCommand(&cobra.Command{
+		Use:   "init",
+		Short: "output the config environment variables and exits",
+		Args:  cobra.NoArgs,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			vars, err := GetVariablesFromConfig(r.prefix, r.cfg)
+			if err != nil {
+				log.WithError(err).Fatal("could not create variables")
 			}
-		}
-		return
+
+			sort.Strings(vars)
+
+			for _, v := range vars {
+				if strings.Contains(v, "=#") {
+					fmt.Println("#" + v)
+				} else {
+					fmt.Println(v)
+				}
+			}
+
+			return nil
+		},
+	})
+
+	return r
+}
+
+func (r *Runner) preRunE(cmd *cobra.Command, args []string) error {
+	envFlag, err := cmd.Flags().GetString("env")
+	if err != nil {
+		return err
 	}
 
-	if err := runtime.Start(runtime.WithMinimumReadMemStatsInterval(time.Second)); err != nil {
-		log.WithError(err).Fatal("could not start runtime tracing")
+	loglevelFlag, err := cmd.Flags().GetString("loglevel")
+	if err != nil {
+		return err
 	}
 
-	ctx, span := tracer.Start(context.Background(), "Run")
-	defer span.End()
+	configFiles, err := cmd.Flags().GetStringSlice("config")
+	if err != nil {
+		return err
+	}
 
 	// First figure out the environment
 	env := Environment(envFlag)
-	span.SetAttributes(attribute.String("env", env.String()))
 
 	// Load the environment from files
 	loadEnvironment(env, configFiles...)
 
 	// Setup logging
-	ctx = setupLogging(ctx, env, loglevelFlag)
+	r.SetupLogging(env, loglevelFlag)
```

Comment:
why ctx no longer needed?

Reply:
No longer setting info in it like was before.